metaflow/flowspec.py

```diff
@@ -133,18 +138,43 @@ def _set_constants(self, kwargs):
                 val = val()
             val = val.split(param.separator) if val and param.separator else val
             setattr(self, var, val)
+            parameters_info.append({"name": var, "type": param.__class__.__name__})
 
         # Do the same for class variables which will be forced constant as modifications
         # to them don't propagate well since we create a new process for each step and
         # re-read the flow file
+        constants_info = []
         for var in dir(self.__class__):
             if var[0] == "_" or var in self._NON_PARAMETERS or var in seen:
                 continue
             val = getattr(self.__class__, var)
             if isinstance(val, (MethodType, FunctionType, property, type)):
                 continue
+            constants_info.append({"name": var, "type": type(val).__name__})
             setattr(self, var, val)
 
+        # We store the DAG information as an artifact called _graph_artifact
+        steps_info, steps_structure = graph.output_steps()
+
+        graph_artifact = {
+            "file": os.path.basename(os.path.abspath(sys.argv[0])),
+            "parameters": parameters_info,
```

Comment:
I am just curious what is the purpose of these parameters here?

Reply:
This is to make explicit what are parameters and what are class-level "constants". Currently, we have no good way of telling these apart and the UI will, for example, display everything that is set at a class level (whether or not it is a parameter) as a Parameter. This will help it address this. It's mostly informational.